lib/phases/cold-launch.js

```diff
@@ -179,6 +180,29 @@ ColdLaunch.prototype.prime = function() {
     });
 };
 
+ColdLaunch.prototype.maybeClearBuffer = function() {
+  if (this.launchesBuffered < Phase.PERFORMANCE_BUFFER_SAFETY) {
+    return Promise.resolve();
+  }
+
+  var phase = this;
+  var marionette = this.device.marionette;
+
+  // Clear the performance buffer after a controlled number of app launches
+  // to keep marking running smoothly, and to avoid hitting the maximum
+  // buffer limit
+  return marionette
+    .startSession()
```

Comment:
I've been thinking about that too, but I'll probably change that in a different bug. Have to consider how to share marionette session without violating concerns.